deploy/kava_testnet/111_deploy_Kava4Pool.ts

```diff
@@ -0,0 +1,67 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+import { MULTISIG_ADDRESSES } from "../../utils/accounts"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts, getChainId } = hre
+  const { execute, get, getOrNull, log, read, save } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  // Manually check if the pool is already deployed
+  const saddleKava2pool = await getOrNull("SaddleKava2pool")
+  if (saddleKava2pool) {
+    log(`reusing "Evmos2poolTokens" at ${saddleKava2pool.address}`)
+  } else {
+    // Constructor arguments
+    const TOKEN_ADDRESSES = [
+      (await get("USDC")).address,
+      (await get("USDT")).address,
+    ]
+    const TOKEN_DECIMALS = [18, 6, 6, 18]
```

Comment:
This will result in reverted tx due array length mismatch (2 vs 4). To prevent errors like this, you can first test the new deployment scripts on hardhat by running

`npx hardhat node --no-deploy`
and then below command on another window
`npx hardhat deploy --deploy-scripts deploy/evmos_testnet/ --network localhost`

Note this will use the evm functionalities of hardhat not the new network. So some modificiations (such as removing `waitForConfirmations` option) to the deploy scripts might be required .